source/content/guides/drupal-9-migration/05-drupal-9-to-pantheon.md

```diff
@@ -0,0 +1,366 @@
+
+This doc uses the following aliases:
+
+- **Alias:** `SITE`
+- **Site Name:** `anita-drupal`
+- **Old site folder** `FORMER-PLATFORM`
+- **Pantheon site folder** `PANTHEON-D9`
+
+### Create a Local Copy of the Old Site's Code
+
+1. Obtain a local copy of your old site's code. Your **code** is all custom and contributed modules or plugins, themes, and libraries. The codebase should not include the `sites/default/files` directory, or any other static assets you do not want tracked by version control.
+
+1. Export the database and media files (`sites/default/files`) from the old platform, but do not add them or upload any files to Pantheon.
+
+### Retrieve a Local Copy of the Pantheon Site's Code
+
+1. From the **<span class="glyphicons glyphicons-wrench"></span> Dev** environment of the Site Dashboard, set the site's Development Mode to Git:
+
+  ![Git connection mode](../../../images/dashboard/connection-mode-git.png)
+
+1. Copy the `git clone` command for the site repository.
+
+  The command should look similar to the following:
+
+  ```bash
+  git clone ssh://codeserver.dev.{site-id}@codeserver.dev.{site-id}.drush.in:2222/~/repository.git
+  ```
+
+1. Run the `git clone` command inside your working folder
+
+
+## Add in the Custom and Contrib Code Needed to Run Your Site
+
+What makes your site code unique is your selection of contributed modules and themes, and any custom modules or themes your development team has created. These customizations need to be replicated in your new project structure.
+
+### Composer packages
+
+1. Copy your package list from the "requires" section of your site's composer.json and add it to the new site's composer.json. If your composer.json defines additional repositories or patches, copy those over too. Take care not to overwrite the `upstream-configuration` package & repository.
+
+  Note: if your old site has custom patches in its codebase, make sure to copy those over as well.
+
+1. `composer update`
+
+1. `git status` to see if files have been added that aren't ignored by `.gitignore`. If anything shows up other than `composer.*`, add it to `.gitignore` until `git status` only shows the composer files being modified.
+
+1. `git add composer.*; git commit -m "Add composer packages"`
+
+  <Accordion title="Repositories and patches in composer.json" id="repositories-and-patches-in-composer-json" icon="info-sign">
+
+  Repository in old site's `composer.json`
+
+  ![repository in old composer.json](https://i.imgur.com/hO0snBW.png)
+
+  Copied to new composer.json without disturbing the "upstream-configuration" one that was already there:
+
+  ![repository moved to new composer.json](https://i.imgur.com/T6eNnXj.png)
+
+  Patches:
+
+  ![repository moved to new composer.json](https://i.imgur.com/x2SYPb1.png)
+
+  </Accordion>
+
+### Custom Code
+
+Manually copy custom code from the old site to the corresponding Pantheon site directory and commit them.
+
+- `web/libraries` is ignored, so `git add -f` if needed.
```

Comment:
The user will have fewer problems if they keep their .gitignore in line with their repository contents. If the user is going to install their libraries using Composer via a project of type `drupal-library`, then they should not add anything to `web/libraries`, and .gitignore can remain unchanged.

If the user wants to commit some libraries directly to `web/libraries`, then they should add an allowlist to the .gitignore file allowing each directory they plan on committing. For example, to commit a directory named foo, add:

```
!/web/libraries/foo
```

Add one line like the one above for each folder committed, changing "foo" as needed.

If the user does not plan on adding any libraries with Composer, they may remove the `web/libraries` line from the .gitignore file completely. If they do this, though, then their build will fail if they ever do add a library with Composer.

Reply:
If you update .gitignore before running `git add` then, `-f` is not required and I think that's maybe what Greg was referring to in his comment.  If the docs reflects this in the right order (update .gitignore, then run `git add`) then the force flag is not required anymore.

Reply:
that makes sense - adjusted to:

> If you commit libraries directly to `web/libraries`, then add each directory to be allowed (not ignored) by `.gitignore`. For example, to commit a `favorite-library` directory, add it and each other directory to the `.gitignore` before you `git add`:
>
> ```none:title=.gitignore
> !/web/libraries/favorite-library
> !/web/libraries/other-favorite-library
> ```